### PR TITLE
Use a DateTime object instead of a timestamp

### PR DESCRIPTION
**Pull request by nickvergessen**
@DeepDiver1975 as per comments discussion DateTime > timestamp

@MorrisJobke @PVince81 @tomneedham

### CODE REVIEW

File: lib/private/notification/inotification.php [hunk 1]
```diff
@@ -61,18 +61,18 @@ public function setUser($user);
 	public function getUser();
 
 	/**
-	 * @param int $timestamp
+	 * @param \DateTime $dateTime
 	 * @return $this
-	 * @throws \InvalidArgumentException if the timestamp are invalid
-	 * @since 8.2.0
+	 * @throws \InvalidArgumentException if the $dateTime is invalid
+	 * @since 9.0.0
```
Comment (PVince81) [inline lib/private/notification/inotification.php]: Version change ? Was that interface available in 8.2 ?

Comment (nickvergessen) [inline lib/private/notification/inotification.php]: It's still in private, so this is fine

**Comment by PVince81** [comment]
:+1:

File: tests/lib/notification/notificationtest.php [hunk 2]
```diff
@@ -139,32 +131,73 @@ public function testSetUserInvalid($user) {
 		$this->notification->setUser($user);
 	}
 
-	public function dataSetTimestamp() {
-		return $this->dataValidInt();
+	public function dataSetDateTime() {
+		$past = new \DateTime();
+		$past->add(new \DateInterval('P1Y'));
```
Comment (rullzer) [inline tests/lib/notification/notificationtest.php]: This seems to be in the future...

File: tests/lib/notification/notificationtest.php [hunk 3]
```diff
@@ -139,32 +131,73 @@ public function testSetUserInvalid($user) {
...
+		$current = new \DateTime();
+		$future = new \DateTime();
+		$future->sub(new \DateInterval('P1Y'));
```
Comment (rullzer) [inline tests/lib/notification/notificationtest.php]: And this the past

**Comment by rullzer** [comment]
Besides my nit picking. :+1:

File: tests/lib/notification/notificationtest.php (hunk 2 above)
Comment (nickvergessen) [inline tests/lib/notification/notificationtest.php]: :hourglass_flowing_sand: 

Fixed

**Comment by MorrisJobke** [comment]
:+1:

**Pull request merged**